toolchain/check/keyword_modifier_set.h

```diff
@@ -16,7 +16,7 @@ LLVM_ENABLE_BITMASK_ENUMS_IN_NAMESPACE();
 
 // The order of modifiers. Each of these corresponds to a group on
 // KeywordModifierSet, and can be used as an array index.
-enum class ModifierOrder : int8_t { Access, Extern, Decl, Last = Decl };
+enum class ModifierOrder : int8_t { Access, Extern, Extend, Decl, Last = Decl };
```

Comment:
If we want to keep the number of modifier groups down, we could put `extern` and `extend` in the same group, since they're mutually exclusive. There's at least a vague semantic connection between them (they both control how the inner declaration adds names to the outer context) -- perhaps enough to justify grouping them? I don't think I have a strong preference either way though.

Reply:
Hm, yeah. I'm not sure. Their similarity in spelling is the worst part, I kept misreading one as the other.

I tried writing a comment explaining the group but I feel like it's a little weird to group them because they happen to be mutually exclusive cuz they are used in entirely different contexts. Slightly lean to keeping them separate.